Ninja and Fortran: ROOT configuration fails on Ubuntu 18.04

Hi @dberzano ,
it looks like that enabling explicitly Fortran (#1218) breaks the configuration of ROOT6 when using the o2-ninja defaults on Ubuntu 18.04 with system ninja.
Here the small log:

```
CMake Error at cmake/modules/CheckCompiler.cmake:15 (enable_language):
  The Ninja generator does not support Fortran using Ninja version

    1.8.2

  due to lack of required features.  Kitware has implemented the required
  features but as of this version of CMake they have not been integrated to
  upstream ninja.  Pending integration, Kitware maintains a branch at:

    https://github.com/Kitware/ninja/tree/features-for-fortran#readme

  with the required features.  One may build ninja from that branch to get
  support for Fortran.
```

Reverting the commit of PR #1218 fix the issue for me, but it cannot work as a general solution.

Ciao,
Max

--- a/ninja.sh
+++ b/ninja.sh
@@ -1,12 +1,9 @@
 package: ninja
-version: "1.8.2"
-tag: "v1.8.2"
-source: https://github.com/ninja-build/ninja
+version: "fortran-%(short_hash)s"
+tag: "5cf291ec1439edffa3c93dde56128181bd5f4037"
+source: https://github.com/Kitware/ninja
 build_requires:
  - "GCC-Toolchain:(?!osx)"
-prefer_system: .*
-prefer_system_check: |
-  ninja --version > /dev/null; if test $? = 127; then exit 1; else case `ninja --version | sed -e 's/.* //' | cut -d. -f1,2,3 | head -n1` in "^(0|2)\.|^1\.([0-7]|[9-99])\.\d*|^1\.8\.[0-1]" ) exit 1 ;; esac; fi; exit 0
 ---
 #!/bin/bash -e
 

